alpa/device_mesh.py

```diff
@@ -908,6 +1009,9 @@ def _launch_xla_servers(self):
                 # "NCCL_LAUNCH_MODE": "PARALLEL",
                 # "XLA_FLAGS": "--xla_dump_to=hlo --xla_dump_hlo_pass_re=.*"
                 # "NCCL_DEBUG": "INFO" if i == 0 else "VERSION",
+                # "NCCL_DEBUG_SUBSYS": "ALL",
+                "NCCL_ALGO": "Ring",
+                "NCCL_PROTO": "Simple",
```

Comment:
Is the CI error and NCCL deadlock due to this change?

Reply:
These two env_vars are essential for broadcast to take advantage of nccl ring to really speed up.

Reply:
And I further debug more deeply and find that the original error is `NCCL WARN Cuda failure 'out of memory'`.

Reply:
![image](https://user-images.githubusercontent.com/45677459/167353925-46581139-30c3-41ad-8be4-2ac7902f0fcc.png)

Reply:
Try to decrease this value, so we leave more memory for other CUDA libraries (e.g., nccl, cublas, cudnn)
https://github.com/alpa-projects/alpa/blob/8a0753db1df49e51db15e823a6cf72d6a32f927b/alpa/global_env.py#L64-L65

Reply:
Thanks!

Reply:
Really appreciate!!! That works!

Reply:
it means broadcast algorithms require more memory?

Reply:
I guess it means broadcast needs larger buffer?

Reply:
#432 can solve this memory issue...

Reply:
Cool, Thanks!

Reply:
After changing from 0.9 to 0.87 for `self.xla_client_mem_fraction`, the above error disappears. 

The following are results for `python3 benchmark_3d.py --suite moe.perf_test_auto`. The first is by broadcast, the second is by send_recv. 

<img width="1046" alt="image" src="https://user-images.githubusercontent.com/45677459/167366776-f3977087-80f3-4123-8986-73fa0a1d9fb7.png">

Broadcast uses less memory. I believe it is because tensor sent by broadcast is more complete, thus more likely to be contiguous, thus requires less extra memcpy in https://github.com/alpa-projects/alpa/blob/3416c2c0ced6f222425867952f355fd514ac1207/alpa/device_mesh.py#L332-L338